source/docs/software/kinematics-and-odometry/intro-and-chassis-speeds.rst

```diff
@@ -9,6 +9,115 @@ The kinematics suite contains classes for differential drive, swerve drive, and
+
+**Position tracking**: Knowing where your robot is on the field allows you to:
+
+- Drive to specific game pieces or scoring locations
+- Dynamically adjust paths based on current position
+- Create reliable autonomous routines that work consistently
+
+**Self-correction**: Combined with vision systems (like AprilTag detection), odometry can be continuously updated to correct for drift and maintain accuracy throughout a match.
+
+### Field-Oriented Control
+
+For holonomic drivetrains (swerve and mecanum), field-oriented control dramatically improves driver experience:
+
+- Driver joystick inputs remain consistent regardless of robot orientation
+- Pushing forward on the joystick always moves toward the opposite alliance station
+- Robot can face any direction while moving in any direction
+- Significantly reduces driver cognitive load during teleoperation
+
+### Trajectory Generation and Following
+
+Kinematics integrates with WPILib's trajectory generation to create smooth, dynamically-constrained paths:
+
+- Generate realistic trajectories that respect your drivetrain's physical limits
+- Follow paths with feedback control for disturbance rejection
+- Combine with path planning tools like PathPlanner or Choreo
+
+### Common Use Cases
+
+**Autonomous routines**: Navigate to game pieces, score at specific locations, execute multi-step sequences
+
+**Vision-assisted driving**: Use AprilTags or other vision targets to update odometry and drive to precise field locations
+
+**Teleoperation enhancement**: Field-oriented control for intuitive driving of holonomic robots
+
+**Match strategy**: Plan routes that avoid defense, optimize cycle times, coordinate with alliance partners
+
+## When to use each drivetrain type
+
+### Differential Drive
+
+**Best for:**
+- Traditional tank drive, "West Coast Drive", or skid-steer robots
+- 6-wheel or 8-wheel configurations with traction wheels
+- Teams new to advanced control systems
+- When pushing power and traction are priorities
+
+**Characteristics:**
+- Cannot strafe (move sideways)
+- Must turn to change direction
+- High traction and pushing power
+- Simpler mechanical design
+- Non-holonomic (requires Ramsete controller for path following)
+
+**Example code**: See the RamseteCommand example in :doc:`/docs/software/examples-tutorials/wpilib-examples`
+
+### Swerve Drive
+
+**Best for:**
+- Maximum maneuverability and control
+- Competitive teams with advanced programming and build experience
+- When field-oriented control is desired
+- Robots that need to move and rotate independently
+
+**Characteristics:**
+- Holonomic (can move in any direction while facing any direction)
+- Highest mobility and control
+- Most complex mechanically and in software
+- Requires precise module control and calibration
+- Can use high-traction wheels
+
+**Example code**: See the SwerveBot and SwerveControllerCommand examples in :doc:`/docs/software/examples-tutorials/wpilib-examples`
```

Comment:
Given wpilibsuite/allwpilib#5098 I'm not sure I'd recommend the WPILib examples.